internal/bind/params.go

```diff
@@ -174,57 +174,44 @@ func toYdbParam(name string, value interface{}) (*params.Parameter, error) {
 	return params.Named(name, v), nil
 }
 
-func Params(args ...interface{}) (parameters []*params.Parameter, _ error) {
-	parameters = make([]*params.Parameter, 0, len(args))
+func Params(args ...interface{}) ([]*params.Parameter, error) {
+	parameters := make([]*params.Parameter, 0, len(args))
 	for i, arg := range args {
+		var newParams []*params.Parameter
+		var err error
 		switch x := arg.(type) {
 		case driver.NamedValue:
-			if x.Name == "" {
-				switch xx := x.Value.(type) {
-				case *params.Parameters:
-					if len(args) > 1 {
-						return nil, xerrors.WithStackTrace(errMultipleQueryParameters)
-					}
-					parameters = *xx
-				case *params.Parameter:
-					parameters = append(parameters, xx)
-				default:
-					x.Name = fmt.Sprintf("$p%d", i)
-					param, err := toYdbParam(x.Name, x.Value)
-					if err != nil {
-						return nil, xerrors.WithStackTrace(err)
-					}
-					parameters = append(parameters, param)
-				}
-			} else {
-				param, err := toYdbParam(x.Name, x.Value)
-				if err != nil {
-					return nil, xerrors.WithStackTrace(err)
-				}
-				parameters = append(parameters, param)
-			}
+			newParams, err = handleNamedValue(x, i, len(args))
 		case sql.NamedArg:
 			if x.Name == "" {
 				return nil, xerrors.WithStackTrace(errUnnamedParam)
 			}
-			param, err := toYdbParam(x.Name, x.Value)
-			if err != nil {
-				return nil, xerrors.WithStackTrace(err)
+			param, pErr := toYdbParam(x.Name, x.Value)
```

Comment:
here you can direct assign

```
newParam, err = toYdbParam(x.Name, x.Value)
```

Why do you need `pErr`? use existed `err` variable